server/server/src/cold_chain/sensor.rs

```diff
@@ -0,0 +1,139 @@
+use actix_web::{
+    http::header,
+    web::{self, Data},
+    HttpRequest, HttpResponse,
+};
+use log::error;
+use mime_guess::mime;
+use regex::Regex;
+use repository::{get_sensor_type, RepositoryError};
+use service::{
+    auth_data::AuthData,
+    sensor::{insert::InsertSensor, update::UpdateSensor},
+    service_provider::ServiceProvider,
+};
+
+use super::validate_request;
+
+#[derive(serde::Deserialize)]
+pub struct Sensor {
+    id: String,
+    #[serde(rename = "macAddress")]
+    mac_address: String,
+    #[serde(rename = "logInterval")]
+    log_interval: i32,
+    #[serde(rename = "programmedDate")]
+    programmed_date: usize,
+    name: String,
+    #[serde(rename = "logDelay")]
+    log_delay: Option<i32>,
+    #[serde(rename = "batteryLevel")]
+    battery_level: i32,
+}
+
+pub async fn sensors(
+    request: HttpRequest,
+    service_provider: Data<ServiceProvider>,
+    auth_data: Data<AuthData>,
+    sensors: web::Json<Vec<Sensor>>,
+) -> HttpResponse {
+    let store_id = match validate_request(request, &service_provider, &auth_data) {
+        Ok((_user, store_id)) => store_id,
+        Err(error) => {
+            let formatted_error = format!("{:#?}", error);
+            return HttpResponse::Unauthorized().body(formatted_error);
+        }
+    };
+    let sensors = sensors.into_inner();
+    if !validate_input(&sensors) {
+        return HttpResponse::BadRequest().body("Expecting a body with the array of sensors");
+    };
+
+    match upsert_sensors(service_provider, store_id, &sensors).await {
+        Ok(response) => response,
+        Err(error) => HttpResponse::InternalServerError().body(format!("{:#?}", error)),
+    }
+}
+
+fn validate_input(sensors: &Vec<Sensor>) -> bool {
+    sensors.into_iter().all(|sensor| validate_sensor(sensor))
+}
+
+fn validate_sensor(sensor: &Sensor) -> bool {
```

Comment:
Nit: for the error case, return an error string (`Result<(), String>`). Then collect all errors in validate_input and return them in the error message...